lib/iris/cube.py

```diff
@@ -3130,18 +3130,19 @@ def _intersect_modulus(
                     pre_wrap_delta != post_wrap_delta
                 )
 
-                # Recalculate the extended minimum.
                 indices = inside_indices[split_cell_indices]
                 cells = bounds[indices]
-                cells_delta = np.diff(coord.bounds[indices])
-
-                # Watch out for ascending/descending bounds
-                if cells_delta[0, 0] > 0:
-                    cells[:, 0] = cells[:, 1] - cells_delta[:, 0]
-                    minimum = np.min(cells[:, 0])
-                else:
-                    cells[:, 1] = cells[:, 0] + cells_delta[:, 0]
-                    minimum = np.min(cells[:, 1])
+                if maximum % modulus not in cells:
```

Comment:
Perhaps a comment here, even referencing the exact problem that this code is aiming to solve.